### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by dev89s**
1. Created class `Classroom` with instance variable `@label` and it has a setter/getter
2. Created "has-many" relationship between relationship between Student & Classroom
3. Adding a student to a classroom sets the classroom for the student also, setting classroom for the student, adds the student to the classroom students
4. Created `Book` class with `@title` and `@author` instance variables and added attr_accessor for the properties
5. Created `Rental` class with `@date`, `@book`, and `@person` attributes. `@date` attribute is initialized in the constructor with getter and setter
6. Added `@retanls` array attribute to `Person` class and set to empty on initialization
7. Added `@rentals` array attribute to `Book` class and set to empty on initialization

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 class Student < Person
-  def initialize(age, classrooom, name = 'Unknown', parent_permission: true)
+  attr_reader :classroom
+
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission:)
-    @classrooom = classrooom
+    @classroom = classroom
+    classroom.add_student(self)
   end
```
Comment (BenMukebo) [inline student.rb]: - [x] When setting the classroom for a student it also adds it to the classrooms' students, kindly let's add a method that serves as the 'belongs-to' side (a student belongs to a classroom). You may check this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`). Your method may look like this ⬇️ 

```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (BenMukebo) [inline book.rb:11]: - [x] The same here. You should have created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. Kindly make sure A method for adding rentals is present.:


```ruby
 def add_rental(person, date)
    ...
  end
```


- Note:  the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

File: person.rb [hunk 3]
```diff
@@ -30,10 +31,10 @@ def correct_name
   end
 end
```
Comment (BenMukebo) [inline person.rb:38]: - [x] The same here, A person has many rentals and A rental belongs to a person. Let's create A method for adding rentals !!

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @dev89s,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Your Pull Request has a proper title and description. ✔️  
- Gitflow was correctly used.  ✔️   
- Professional README ✔️  
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @dev89s, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested change implemented ✅ 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**